src/logic/function/basic.lean

```diff
@@ -671,6 +671,9 @@ by rw [apply_ite f, h, ite_not]
 protected lemma eq_iff {x y : α} : f x = y ↔ x = f y :=
 h.injective.eq_iff' (h y)
 
+protected lemma inj {x y : α} : f x = f y ↔ x = y :=
+h.injective.eq_iff
```

Comment:
I'm not really convinced by this lemma; the name is a bit weird since the corresponding lemma for `injective` is not `injective.inj`, and `.injective.eq_iff` isn't all that arduous a spelling.